pkg/prometheus/promcfg.go

```diff
@@ -727,6 +731,16 @@ func (cg *ConfigGenerator) generatePodMonitorConfig(
 	cfg = enforcer.addLimitsToYAML(cfg, labelNameLengthLimitKey, m.Spec.LabelNameLengthLimit, enforcedLabelNameLengthLimit)
 	cfg = enforcer.addLimitsToYAML(cfg, labelValueLengthLimitKey, m.Spec.LabelValueLengthLimit, enforcedLabelValueLengthLimit)
 
+	// Since BodySizeLimit is defined only in PrometheusCRD
+	if enforcedBodySizeLimit != "" {
+		if version.Major == 2 && version.Minor >= 28 {
+			cfg = append(cfg, yaml.MapItem{Key: "body_size_limit", Value: enforcedBodySizeLimit})
+		} else {
+			level.Warn(cg.logger).Log("msg", "PodMonitor 'body_size_limit' is only available from prometheus 2.28.",
+				"podMonitor", m.Name)
```

Comment:
`logger` is already decorated with `podMonitor` and `namespace` keys. 

```suggestion
			level.Warn(logger).Log("msg", "'body_size_limit' is only available from prometheus 2.28.",
				"version", version)
```